satgpt.py

```diff
@@ -92,14 +92,16 @@ def query_chatbot():
     try:
         # Parse the request data
         data = request.get_json()
-
+        print("here is the data in TRY ", data)
         # if query is not in data, return error
-        if "query" and "user_uuid" not in data and "r_hash" not in data:
+        if "query" not in data and "user_uuid" not in data and "r_hash" not in data:
```

Comment:
I'm not sure if I had the previous logic correct, but this isn't exactly what I want either. I want the query to return error if `query` and `user_uuid` BOTH are not in data OR `r_hash` is not in data. If `query` is in but `user_uuid` isn't or vice versa, this will fail (see line 156/157). Really, the case of `r_hash` is checking for a payment and the case of `query`/`user_uuid` are creating an invoice to be paid, so they are mutually exclusive and we should make these into separate queries with separate messages, but we can save that for another time unless you want to tackle it

Reply:
Hmm, I'm still slightly confused. Is this what we want -> 
First, check if query and uuid is in data. They must both be present, if not we return error "no query provided".
If they are both present, we next check if r_hash is present. If it is, then we check if payment has been made.
If r-hash is not present, we create a new invoice and r_hash. 

If all of that sounds correct, then I think this line should just read if "query" not in data or "user_uuid" not in data:". We already check for the r_hash in the next elif statement.

Reply:
Yeah so this is definitely confusing haha, but basically the first check there is checking whether the request was _malformed_. It is malformed if it doesn't have everything needed to create a proper response (it needs a `query` param to know which question it's asking and a `user_uuid` param to know who to charge), OR if it's simply checking for a payment that was already made, in which case an `r_hash` is needed. If neither of the following conditions are met:

1) the request contains `query` and `user_uuid` fields
OR
2) the request contains an `r_hash` field

Then the query is malformed because we can't do anything with whatever other info they gave us, so we return a `400 Bad Request` error.

If it wasn't malformed, then we're checking _which_ of the query params are present (since it could either be saving a query and user payment details using `query` and `user_uuid` or checking for a payment using `r_hash`), so we need to do the `if` statements again to see _what_ we're doing with this api call.

Really, the confusion here is due to poor design on my part, since these should be two separate api endpoints, one for saving the payment info & query and one for polling to check whether a payment has been made using the `r_hash`. We'll have to do that eventually, so let's do that in the next PR.

Reply:
I see what you mean. But I thought whenever an invoice has been paid, the data sent to the function now contains the new r_hash AND query + user_uuid, not just the r_hash?
We might need to hop on a call to discuss further...